resdk/scripts.py

```diff
@@ -209,52 +212,65 @@ def parse_annotation_file(annotation_file):
 
-        if read_schema:
-            descriptor_schema = read_schema['slug']
-            descriptor = {
-                'barcode': annotations[fn].get('BARCODE', None),
-                'barcode_removed': True if annotations[fn].get('BARCODE_REMOVED', 'N').upper() == 'Y' else False,
-                'instrument_type': annotations[fn].get('INSTRUMENT', None),
-                'seq_date': annotations[fn].get('SEQ_DATE', None),
-            }
-
-        # Paired-end reads
-        if annotations[fn]['PAIRED_END'] == 'Y' and annotations[fn]['FASTQ_PATH_PAIR']:
-            slug = 'upload-fastq-paired'
-            input_['src1'] = [fn]
-            input_['src2'] = [os.path.join(genialis_seq_dir, annotations[fn]['FASTQ_PATH_PAIR'])]
-
-        # Single-end reads
-        else:
-            slug = 'upload-fastq-single'
-            input_['src'] = [fn]
-
-        data = resolwe.run(slug, input_, descriptor, descriptor_schema, data_name=annotations[fn]['SAMPLE_NAME'])
-
-        if data:
-            uploaded_files.append(fn)
-
-            presample = resolwe.api.presample.get(data=data.id)[0]
-
-            if 'geo' not in presample['descriptor']:
-                presample['descriptor']['geo'] = {}
-
-            organism = ORGANISMS.get(annotations[fn]['ORGANISM'].upper(), '')
-            if organism:
-                presample['descriptor']['geo']['organism'] = organism
-
-            experiment_type = EXPERIMENT_TYPE.get(annotations[fn]['SEQ_TYPE'].upper(), '')
-            if experiment_type:
-                presample['descriptor']['geo']['experiment_type'] = experiment_type
-
-            resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor']})
-
-        else:
-            print("Error uploading {}".format(fn), file=sys.stderr)
+    for sample_n in annotations.keys():
+        input_ = {}
+        fw_reads = annotations[sample_n]['FASTQ_PATH'].split(',')
+
+        if set(fw_reads).issubset(set(all_new_read_files_uploaded)):
+            descriptor, descriptor_schema = None, None
+
+            if read_schema:
+                descriptor_schema = read_schema['slug']
+                descriptor = {
+                    'barcode': annotations[sample_n].get('BARCODE', None),
+                    'barcode_removed': True if annotations[sample_n].get('BARCODE_REMOVED', 'N').upper() == 'Y'
+                                       else False,
+                    'instrument_type': annotations[sample_n].get('INSTRUMENT', None),
+                    'seq_date': annotations[sample_n].get('SEQ_DATE', None),
+                }
+
+            # Paired-end reads
+            if annotations[sample_n]['PAIRED_END'] == 'Y' and annotations[sample_n]['FASTQ_PATH_PAIR']:
+                rw_reads = annotations[sample_n]['FASTQ_PATH_PAIR'].split(',')
+                slug = 'upload-fastq-paired'
+                input_['src1'] = fw_reads
+                input_['src2'] = [os.path.join(genialis_seq_dir, f) for f in rw_reads]
```

Comment:
Add also os.path.normath() on top of os.path.join, to avoid invalid paths like "/path/to/here/../here/file.fq"